packages/components/PrefixBadgePicker/PrefixBadgePicker.tsx

```diff
@@ -29,153 +21,57 @@ export function PrefixBadgePicker({
 
   /** Callback when a badge has been selected. */
   onSelect?: (prefixBadge?: PrefixBadge) => void;
-
-  /** Control the opened or closed state. Useful for testing. */
-  initialState?: States;
 }) {
-  const id = (React as any).useId();
   const [activeBadge, setActiveBadge] = React.useState<PrefixBadge | undefined>(initialValue);
-  const [state, setState] = React.useState<States>(initialState);
-  const isOpen = state === "opening" || state === "opened";
-
-  const handleSelect = (prefixBadgeName?: PrefixBadge) => {
-    setState("closed");
+  const handleSelect = (prefixBadgeName: PrefixBadge | undefined) => {
     setActiveBadge(prefixBadgeName);
     onSelect?.(prefixBadgeName);
   };
 
   return (
-    <motion.div
-      className={classNames(styles.PrefixBadgePicker, { [styles.isOpen]: isOpen })}
-      whileHover="parentHover"
-    >
-      <AnimatePresence>
-        <motion.div
-          key="fill"
-          initial={false}
-          animate={{
-            width: isOpen ? openedWidth : closedWidth,
-            boxShadow: isOpen ? shadowActive : shadowInitial,
-          }}
-          transition={{
-            duration,
-            delay: isOpen ? 0 : duration * 3,
-            boxShadow: { duration: 0.2, delay: 0 },
-          }}
-          variants={{
-            parentHover: {
-              boxShadow: shadowActive,
-            },
-          }}
-          className={styles.PrefixBadgePickerFill}
-        />
-
-        {(activeBadge === undefined || isOpen) && (
+    <div className="z-10 flex" style={{ width: 26, height: 26 }}>
+      <Picker<PrefixBadge | undefined>
+        value={activeBadge}
+        onChange={handleSelect}
+        className={styles.PrefixBadgePicker}
+      >
+        <ToggleButton id="toggle" />
+        {badges.map(badge => (
           <motion.button
-            key="trigger"
-            type="button"
-            initial={false}
-            animate={{ opacity: 1, scale: 1 }}
-            exit={{ opacity: 0, scale: 0.9 }}
-            className={styles.PrefixBadge}
-            onKeyDown={(event: React.KeyboardEvent) => {
-              if (event.key === "Enter") {
-                if (state === "closed") {
-                  setState("opening");
-                } else {
-                  handleSelect();
-                }
-              }
-            }}
-            onMouseDown={() => {
-              if (state === "closed") {
-                setState("opening");
-              }
-            }}
-            onMouseUp={() => {
-              if (state === "opened") {
-                handleSelect();
-              } else {
-                setState("opened");
-              }
```

Comment:
This mouse down interaction was causing issues with the click interaction and keyboard so I'd like to revisit this and probably use a library so it's accessible.